ocaml/lambda/matching.ml

```diff
@@ -1799,10 +1799,12 @@ let get_expr_args_constr ~scopes head (arg, _mut, sort, layout) rem =
   Array.iter (fun jkind ->
       jkind_layout_default_to_value_and_check_not_void head.pat_loc jkind)
     cstr.cstr_arg_jkinds;
+  let ubr = Translmode.transl_unique_barrier (head.pat_unique_barrier) in
```

Comment:
If the unique barrier is stored in individual pattern descriptors instead of in the pattern heaer, this line gets harder. My guess is that, here, storing the unique barrier in the `Construct` constructor of `Paterns.Head.desc` may work (and similarly for other cases).